# Introduction


In today's digital era, online shopping has completely transformed how we buy and connect with businesses. It's amazing how a simple click lets customers explore, purchase, and even seek help for products or services online. One really important part of this online experience is how satisfied customers are with the support they receive – it's a big deal because it shapes how they feel about a brand overall and whether they'll stick around as loyal customers.

When we dig into relevant data, we can learn so much about how customers behave, what they like, and where they struggle with support services. This kind of information helps us spot areas where we can do better, make processes more efficient, and ultimately make customers happier.

In [363]:

import numpy as np # linear algebra
import pandas as pd # data processing

#Visualization
import matplotlib.pyplot as plt
import seaborn as sns

# Import packages for data preprocessing
from sklearn import metrics
from sklearn.preprocessing import OneHotEncoder
from sklearn.feature_extraction.text import CountVectorizer, TfidfVectorizer
from sklearn.utils import resample

# Import packages for data modeling
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import classification_report
from sklearn.metrics import confusion_matrix, ConfusionMatrixDisplay

#warnings
from warnings import filterwarnings
import warnings
warnings.simplefilter(action='ignore')

# OVERVIEW

 This comprehensive dataset captures customer satisfaction scores over a one-month period at the Shopzilla e-commerce platform, a pseudonymous entity. With 85,907 rows and 20 columns, this dataset provides a rich source for conducting Exploratory Data Analysis (EDA), Visualization, and Machine Learning Classification tasks. 

# Purpose

This dataset serves as a robust resource for evaluating customer service performance, forecasting satisfaction levels, and conducting customer behavior analysis within the e-commerce sector. The information contained in the dataset includes crucial features such as channel name, order details, customer feedback, agent information, and, most importantly, Customer Satisfaction (CSAT) scores.

Purpose of CSAT
CSAT helps organizations gauge how well they are meeting customer expectations. It identifies areas for improvement and measures the effectiveness of customer service, product quality, or process efficiency.

# Data Description:¶


Unique id: Unique identifier for each record

Channel name: Name of the customer service channel

Category: Category of the interaction

Sub-category: Sub-category of the interaction

Customer Remarks: Feedback provided by the customer

Order id: Identifier for the order associated with the interaction

Order date time: Date and time of the order

Issue reported at: Timestamp when the issue was reported

Issue responded: Timestamp when the issue was responded to

Survey response date: Date of the customer survey response

Customer city: City of the customer

Product category: Category of the product

Item price: Price of the item

Connected handling time: Time taken to handle the interaction

Agent name: Name of the customer service agent

Supervisor: Name of the supervisor

Manager: Name of the manager

Tenure Bucket: Bucket categorizing agent tenure

Agent Shift: Shift timing of the agent

CSAT Score: Customer Satisfaction (CSAT) score

In [370]:
data=pd.read_csv('Customer_support_data.csv')

# Data Exploration

In [372]:
data.head()

Unique id channel_name         category  \
0  7e9ae164-6a8b-4521-a2d4-58f7c9fff13f      Outcall  Product Queries   
1  b07ec1b0-f376-43b6-86df-ec03da3b2e16      Outcall  Product Queries   
2  200814dd-27c7-4149-ba2b-bd3af3092880      Inbound    Order Related   
3  eb0d3e53-c1ca-42d3-8486-e42c8d622135      Inbound          Returns   
4  ba903143-1e54-406c-b969-46c52f92e5df      Inbound     Cancellation   

                   Sub-category Customer Remarks  \
0                Life Insurance              NaN   
1  Product Specific Information              NaN   
2             Installation/demo              NaN   
3        Reverse Pickup Enquiry              NaN   
4                    Not Needed              NaN   

                               Order_id order_date_time Issue_reported at  \
0  c27c9bb4-fa36-4140-9f1f-21009254ffdb             NaN  01/08/2023 11:13   
1  d406b0c7-ce17-4654-b9de-f08d421254bd             NaN  01/08/2023 12:52   
2  c273368d-b961-44cb-beaf-62d6fd6c00d5             NaN  01/08/2023 20:16   
3  5aed0059-55a4-4ec6-bb54-97942092020a             NaN  01/08/2023 20:56   
4  e8bed5a9-6933-4aff-9dc6-ccefd7dcde59             NaN  01/08/2023 10:30   

    issue_responded Survey_response_Date Customer_City Product_category  \
0  01/08/2023 11:47            01-Aug-23           NaN              NaN   
1  01/08/2023 12:54            01-Aug-23           NaN              NaN   
2  01/08/2023 20:38            01-Aug-23           NaN              NaN   
3  01/08/2023 21:16            01-Aug-23           NaN              NaN   
4  01/08/2023 10:32            01-Aug-23           NaN              NaN   

   Item_price  connected_handling_time           Agent_name      Supervisor  \
0         NaN                      NaN     Richard Buchanan     Mason Gupta   
1         NaN                      NaN        Vicki Collins       Dylan Kim   
2         NaN                      NaN         Duane Norman    Jackson Park   
3         NaN                      NaN       Patrick Flores     Olivia Wang   
4         NaN                      NaN  Christopher Sanchez  Austin Johnson   

           Manager    Tenure Bucket Agent Shift  CSAT Score  
0  Jennifer Nguyen  On Job Training     Morning           5  
1      Michael Lee              >90     Morning           5  
2      William Kim  On Job Training     Evening           5  
3       John Smith              >90     Evening           5  
4      Michael Lee             0-30     Morning           5

In [373]:
data.tail()

Unique id channel_name        category  \
85902  505ea5e7-c475-4fac-ac36-1d19a4cb610f      Inbound  Refund Related   
85903  44b38d3f-1523-4182-aba2-72917586647c      Inbound   Order Related   
85904  723bce2c-496c-4aa8-a64b-ca17004528f0      Inbound   Order Related   
85905  707528ee-6873-4192-bfa9-a491f1c08ab5      Inbound        Feedback   
85906  07c7a878-0d5a-42e0-97ef-de59abec0238      Inbound         Returns   

                   Sub-category                          Customer Remarks  \
85902            Refund Enquiry                                       NaN   
85903    Seller Cancelled Order  Supported team customer executive   good   
85904      Order status enquiry      need to improve with proper details.   
85905  UnProfessional Behaviour                                       NaN   
85906    Reverse Pickup Enquiry                                       NaN   

                                   Order_id order_date_time Issue_reported at  \
85902  1b5a2b9c-a95f-405f-a42e-5b1b693f3dc9             NaN  30/08/2023 23:20   
85903  d0e8a817-96d5-4ace-bb82-adec50398e22             NaN  31/08/2023 08:15   
85904  bdefe788-ccec-4eda-8ca4-51045e68db8a             NaN  31/08/2023 18:57   
85905  a031ec28-0c5e-450e-95b2-592342c40bc4             NaN  31/08/2023 19:59   
85906  3230db30-f8da-4c44-8636-ec76d1d3d4f3             NaN  31/08/2023 23:36   

        issue_responded Survey_response_Date Customer_City Product_category  \
85902  31/08/2023 07:22            31-Aug-23           NaN              NaN   
85903  31/08/2023 08:17            31-Aug-23           NaN              NaN   
85904  31/08/2023 19:02            31-Aug-23           NaN              NaN   
85905  31/08/2023 20:00            31-Aug-23           NaN              NaN   
85906  31/08/2023 23:37            31-Aug-23           NaN              NaN   

       Item_price  connected_handling_time         Agent_name     Supervisor  \
85902         NaN                      NaN       Brandon Leon      Ethan Tan   
85903         NaN                      NaN       Linda Foster     Noah Patel   
85904         NaN                      NaN  Kimberly Martinez    Aiden Patel   
85905         NaN                      NaN      Daniel Martin  Olivia Suzuki   
85906         NaN                      NaN   Elizabeth Guerra   Nathan Patel   

               Manager    Tenure Bucket Agent Shift  CSAT Score  
85902      William Kim  On Job Training     Morning           4  
85903       Emily Chen              >90     Morning           5  
85904       Olivia Tan  On Job Training     Evening           5  
85905       Olivia Tan              >90     Morning           4  
85906  Jennifer Nguyen  On Job Training     Evening           5

In [374]:
display(data[0:3].T)

0  \
Unique id                7e9ae164-6a8b-4521-a2d4-58f7c9fff13f   
channel_name                                          Outcall   
category                                      Product Queries   
Sub-category                                   Life Insurance   
Customer Remarks                                          NaN   
Order_id                 c27c9bb4-fa36-4140-9f1f-21009254ffdb   
order_date_time                                           NaN   
Issue_reported at                            01/08/2023 11:13   
issue_responded                              01/08/2023 11:47   
Survey_response_Date                                01-Aug-23   
Customer_City                                             NaN   
Product_category                                          NaN   
Item_price                                                NaN   
connected_handling_time                                   NaN   
Agent_name                                   Richard Buchanan   
Supervisor                                        Mason Gupta   
Manager                                       Jennifer Nguyen   
Tenure Bucket                                 On Job Training   
Agent Shift                                           Morning   
CSAT Score                                                  5   

                                                            1  \
Unique id                b07ec1b0-f376-43b6-86df-ec03da3b2e16   
channel_name                                          Outcall   
category                                      Product Queries   
Sub-category                     Product Specific Information   
Customer Remarks                                          NaN   
Order_id                 d406b0c7-ce17-4654-b9de-f08d421254bd   
order_date_time                                           NaN   
Issue_reported at                            01/08/2023 12:52   
issue_responded                              01/08/2023 12:54   
Survey_response_Date                                01-Aug-23   
Customer_City                                             NaN   
Product_category                                          NaN   
Item_price                                                NaN   
connected_handling_time                                   NaN   
Agent_name                                      Vicki Collins   
Supervisor                                          Dylan Kim   
Manager                                           Michael Lee   
Tenure Bucket                                             >90   
Agent Shift                                           Morning   
CSAT Score                                                  5   

                                                            2  
Unique id                200814dd-27c7-4149-ba2b-bd3af3092880  
channel_name                                          Inbound  
category                                        Order Related  
Sub-category                                Installation/demo  
Customer Remarks                                          NaN  
Order_id                 c273368d-b961-44cb-beaf-62d6fd6c00d5  
order_date_time                                           NaN  
Issue_reported at                            01/08/2023 20:16  
issue_responded                              01/08/2023 20:38  
Survey_response_Date                                01-Aug-23  
Customer_City                                             NaN  
Product_category                                          NaN  
Item_price                                                NaN  
connected_handling_time                                   NaN  
Agent_name                                       Duane Norman  
Supervisor                                       Jackson Park  
Manager                                           William Kim  
Tenure Bucket                                 On Job Training  
Agent Shift                                           Evening  
CSAT Score                                      

In [375]:
data.shape

(85907, 20)

In [376]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 85907 entries, 0 to 85906
Data columns (total 20 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Unique id                85907 non-null  object 
 1   channel_name             85907 non-null  object 
 2   category                 85907 non-null  object 
 3   Sub-category             85907 non-null  object 
 4   Customer Remarks         28742 non-null  object 
 5   Order_id                 67675 non-null  object 
 6   order_date_time          17214 non-null  object 
 7   Issue_reported at        85907 non-null  object 
 8   issue_responded          85907 non-null  object 
 9   Survey_response_Date     85907 non-null  object 
 10  Customer_City            17079 non-null  object 
 11  Product_category         17196 non-null  object 
 12  Item_price               17206 non-null  float64
 13  connected_handling_time  242 non-null    float64
 14  Agent_name            

In [377]:
data.dtypes

Unique id                   object
channel_name                object
category                    object
Sub-category                object
Customer Remarks            object
Order_id                    object
order_date_time             object
Issue_reported at           object
issue_responded             object
Survey_response_Date        object
Customer_City               object
Product_category            object
Item_price                 float64
connected_handling_time    float64
Agent_name                  object
Supervisor                  object
Manager                     object
Tenure Bucket               object
Agent Shift                 object
CSAT Score                   int64
dtype: object

Based on data set info. This df has dumped with 85907 rows and 20 Columns.

Except 3 variable such as (Item_price, Connected_handling_time and CSAT Score) all other variables seems to be Object data type.

The data shows like non-null but as per count in some variable, many data might be fill with NAN, NA or some other charecters.

### It has 20 columns, each representing a specific aspect of the interactions, and 85,907 rows

## Make a copy of the orginal Dataset

In [381]:
df=data.copy()

### Covert titles of Columns to Lower case

In [383]:
def change_columns_to_lowercase(df):
    # Get the current column names
    old_column=df.columns.tolist()
    # Convert column names to lowercase
    new_col=[col.lower()for col in old_column]
    # Rename columns in the DataFrame
    df.columns = new_col
    return df


# Apply the function to change column names to lowercase

df=change_columns_to_lowercase(df)   

### Make the Space between the column names with underscore

In [385]:
def replace_space_with_underscore(df):
    old_columns=df.columns.tolist()
    new_col=[col.replace(' ','_') for col in old_columns]
    df.columns=new_col
    return df

# Apply the function to replace spaces with underscores
df= replace_space_with_underscore(df)    

In [386]:
df.columns

Index(['unique_id', 'channel_name', 'category', 'sub-category',
       'customer_remarks', 'order_id', 'order_date_time', 'issue_reported_at',
       'issue_responded', 'survey_response_date', 'customer_city',
       'product_category', 'item_price', 'connected_handling_time',
       'agent_name', 'supervisor', 'manager', 'tenure_bucket', 'agent_shift',
       'csat_score'],
      dtype='object')

In [387]:
#finding duplicate datas
df.duplicated().sum()

0

## Understand each columns


In [389]:
cat=[]
cat=[col for col in df.columns if df[col].dtypes=='object']

In [390]:
cat

['unique_id',
 'channel_name',
 'category',
 'sub-category',
 'customer_remarks',
 'order_id',
 'order_date_time',
 'issue_reported_at',
 'issue_responded',
 'survey_response_date',
 'customer_city',
 'product_category',
 'agent_name',
 'supervisor',
 'manager',
 'tenure_bucket',
 'agent_shift']

In [391]:
for i in cat:  
    print(i)
    print(df[i].value_counts())

unique_id
unique_id
7e9ae164-6a8b-4521-a2d4-58f7c9fff13f    1
4a7fec30-68f5-4891-b934-6a666e00930e    1
67792991-fd16-465f-bbdc-3189a0f16395    1
c629cb55-3c2b-448c-8598-96eeb0f30f13    1
67f2636b-055f-4ae5-a360-c3f35c09205c    1
                                       ..
71c4d872-ed1a-4215-98fc-0753046c7637    1
8ae06ce2-ffd7-4c5f-9820-56226ec30275    1
52b3f399-ee61-41fa-b332-d830ab176977    1
4c9be137-c6ea-48ea-b80b-5b5dda67110d    1
07c7a878-0d5a-42e0-97ef-de59abec0238    1
Name: count, Length: 85907, dtype: int64
channel_name
channel_name
Inbound    68142
Outcall    14742
Email       3023
Name: count, dtype: int64
category
category
Returns               44097
Order Related         23215
Refund Related         4550
Product Queries        3692
Shopzilla Related      2792
Payments related       2327
Feedback               2294
Cancellation           2212
Offers & Cashback       480
Others                   99
App/website              84
Onboarding related       65
Name: count, dtype: 

Onboarding" typically refers to the initial setup or enrollment process when a customer or partner starts using a company's services, products, or platform. It’s the experience of guiding them through the first steps.

High-Frequency Sub-Categories:
Reverse Pickup Enquiry (22,389 cases):

Customers inquiring about the process, timing, or status of a reverse pickup for a product they want to return.
Return Request (8,523 cases):

Requests to return a purchased item due to dissatisfaction, defects, or other reasons.
Delayed (7,388 cases):

Complaints regarding delays in delivery, service, or order processing.
Order Status Enquiry (6,922 cases):

Customers asking for updates on their order status (e.g., "Where is my package?").
Installation/Demo (4,116 cases):

Requests for assistance in setting up a product or for a demonstration (e.g., appliances, electronics).
Mid-Frequency Sub-Categories:
Fraudulent User (4,108 cases):

Issues reported about suspected fraudulent activities, like scams or misuse of the platform.
Product Specific Information (3,589 cases):

Questions about product details, such as features, specifications, or compatibility.
Refund Enquiry (2,665 cases):

Customers asking about the status or process of getting a refund.
Wrong (2,597 cases):

Cases where the customer received an incorrect item or service.
Missing (2,556 cases):

Items or components that were missing from a package or order.
Other Significant Sub-Categories:
UnProfessional Behaviour (2,294 cases):

Complaints about the behavior of staff or agents.
Service Centres Related (1,875 cases):

Inquiries about or issues with service centers, such as availability or delays.
Invoice Request (1,465 cases):

Requests for invoices or proof of purchase.
Online Payment Issues (1,079 cases):

Problems related to payments made online, like failed transactions or deductions without confirmations.
Exchange/Replacement (896 cases):

Requests for exchanging or replacing a product.
Specialized or Low-Frequency Sub-Categories:
Damaged (475 cases):

Products reported as damaged upon delivery.
Technician Visit (378 cases):

Scheduling or issues related to a technician’s visit for repairs or installations.
Wallet Related (230 cases):

Problems involving the company’s digital wallet (e.g., balance errors or usability issues).
PayLater Related (140 cases):

Queries about the "Pay Later" feature, like eligibility, limits, or pending payments.
COD Refund Details (85 cases):

Refund-related queries specifically for Cash-on-Delivery orders.
Warranty Related (78 cases):

Questions or complaints about product warranties.
Rare Issues:
Life Insurance (23 cases):

Support requests related to life insurance products or add-ons.
Others (12 cases):

Miscellaneous issues that don’t fit into other categories.
Unable to Login (7 cases):

Login-related problems, such as forgotten passwords or account lockouts.
Commission Related (3 cases):

Inquiries about commissions, likely related to partner or affiliate programs.
Policy Related (2 cases):

Questions or disputes about company policies, such as returns or refunds.
Non-Order Related (1 case):

A single case unrelated to any orders, likely a general or unique issue.




In [393]:
data.iloc[:,4].value_counts()[:10]

Customer Remarks
Good          1390
Good          1158
Very good      569
Nice           316
Thanks         276
Ok             259
No             258
Thank you      244
Nice           239
Very good      236
Name: count, dtype: int64

In [394]:
def summary(df):
    print(f'data shape:{df.shape}')
    # Get data type
    dtype_df=pd.DataFrame(df.dtypes,columns=['Data Type'])
    ## Count missing values
    missing_values_df=pd.DataFrame(df.isnull().sum(),columns=['#missing'])
    missing_values_df['%missing'] = df.isnull().mean() * 100
    # Count unique values
    unique_values_df = pd.DataFrame(df.nunique(), columns=['#unique'])
    # Describe statistics
    describe_df = df.describe(include='all').transpose()
    stats_df = describe_df[['top', 'freq', 'min', 'max', 'mean', 'std']]
     # Count duplicate rows
    duplicate_rows = df.duplicated().sum()
    # Concatenate all dataframes
    summary_df =pd.concat([dtype_df, missing_values_df,unique_values_df,stats_df],axis=1)
    summary_df.loc['#duplicate'] = duplicate_rows
    styled_summary = summary_df.style.background_gradient(cmap='GnBu')
    return styled_summary

summary(df)

data shape:(85907, 20)


What we know so far:

More than 99% connected_handling_time is missing so this column could be removed and we are not able to do any analysis based on that

Majority of order_date_time, Customer City, Product_category, Customer Remarks,item_price are also missing to. So if we removed missing observations(rows) related to those missing values, we will lose significant part of our dataset and there would be not much to Analyzing

Based on what we have, let's keep some columns and and remove those with huge missing values( remove columns not rows, axis=1)

In [396]:
df.describe().T

count         mean           std  min    25%  \
item_price               17206.0  5660.774846  12825.728411  0.0  392.0   
connected_handling_time    242.0   462.400826    246.295037  0.0  293.0   
csat_score               85907.0     4.242157      1.378903  1.0    4.0   

                           50%      75%       max  
item_price               979.0  2699.75  164999.0  
connected_handling_time  427.0   592.25    1986.0  
csat_score                 5.0     5.00       5.0

Describe is actually an 5 point summary which will calculate for numerical values. So the reason only 3 variable got described.

Item_price and connected_handling_time has its bell shape curve with a huge left skewness. and these both seems to have too many outlier, which can be clearly explained in visualization.mean>median

CSAT variable has numerical Dtype but it looks like a categorical variable.


# DATA CLEANING

By looking above analysis we could notice categories such as Unique ID, Customer Remark and Order_ID wont be usefull for further analysis so we suppose to drope those Variables.
By looking above analysis we could notice categories such as Unique ID, Customer Remark and Order_ID wont be usefull for further analysis so we suppose to drope those Variables.

Some object variables looks to be like time stamp columns. So the data type need to get change for those variables.

Some Object and float dtype variables has filled with NAN. Those needs to get filled as per needs.

In [400]:
df.columns

Index(['unique_id', 'channel_name', 'category', 'sub-category',
       'customer_remarks', 'order_id', 'order_date_time', 'issue_reported_at',
       'issue_responded', 'survey_response_date', 'customer_city',
       'product_category', 'item_price', 'connected_handling_time',
       'agent_name', 'supervisor', 'manager', 'tenure_bucket', 'agent_shift',
       'csat_score'],
      dtype='object')

In [401]:
#Dropping not needed variables
df1=df.drop(["unique_id","customer_remarks","order_id","order_date_time"],axis=1)
df1.sample(n=2)

#Replacing object dtype to timestamp dtype
timestamp_columns=['issue_reported_at','issue_responded','survey_response_date']
for col in timestamp_columns:
    df1[col]=pd.to_datetime(df1[col],dayfirst=True,errors='coerce')
    print(col,": dtype =",df1[col].dtype)
    
#In the context of the pd.to_datetime() function in Pandas, the dayfirst=True argument specifies the format in which the day comes before the month when parsing dates.
# When using pd.to_datetime(), some values in the data might not be valid dates or timestamps. By setting errors='coerce', 
#any invalid parsing will result in those entries being converted to
#NaT (Not a Time), which is pandas' way of representing missing or null values in datetime objects.

issue_reported_at : dtype = datetime64[ns]
issue_responded : dtype = datetime64[ns]
survey_response_date : dtype = datetime64[ns]


In [402]:
#Replacing null value according to needs
for i in df1.columns:
# Impute missing values for categorical features-customer city and product category
    if (df1[i].isnull().sum()>0 and df1[i].dtype=="object"):
        df1[i].fillna("Unknown",inplace=True)
        print(i, "Variable has object dtype, so it got Replace its null values to Unknown \n")
## Impute missing values for numerical features        
    elif (df1[i].isnull().sum()>0 and df1[i].dtype=="float64"):
        df1[i].fillna(df1[i].median(),inplace=True)
        print(i, "Variable has float dtype, so it got Replace its null values to columns median value \n")

customer_city Variable has object dtype, so it got Replace its null values to Unknown 

product_category Variable has object dtype, so it got Replace its null values to Unknown 

item_price Variable has float dtype, so it got Replace its null values to columns median value 

connected_handling_time Variable has float dtype, so it got Replace its null values to columns median value 



In [407]:
# finding null values and unique value counts 
null_values1= round(df1.isnull().mean()*100)
unique_values1=df1.nunique()
analysis_df1 = pd.DataFrame({'column_name': df1.columns,"unique_values": unique_values1,'null_values_%': null_values1,"data_type":df1.dtypes})
analysis_df1.reset_index(drop=True,inplace=True)
analysis_df1
#null_values1= round(df1.isnull().sum() * 100 / len(df),2)

column_name  unique_values  null_values_%       data_type
0              channel_name              3            0.0          object
1                  category             12            0.0          object
2              sub-category             57            0.0          object
3         issue_reported_at          30923            0.0  datetime64[ns]
4           issue_responded          30262            0.0  datetime64[ns]
5      survey_response_date             31            0.0  datetime64[ns]
6             customer_city           1783            0.0          object
7          product_category             10            0.0          object
8                item_price           2789            0.0         float64
9   connected_handling_time            211            0.0         float64
10               agent_name           1371            0.0          object
11               supervisor             40            0.0          object
12                  manager              6            0.0          object
13            tenure_bucket              5            0.0          object
14              agent_shift              5            0.0          object
15               csat_score              5            0.0           int64

In [405]:
df1

channel_name         category                  sub-category  \
0          Outcall  Product Queries                Life Insurance   
1          Outcall  Product Queries  Product Specific Information   
2          Inbound    Order Related             Installation/demo   
3          Inbound          Returns        Reverse Pickup Enquiry   
4          Inbound     Cancellation                    Not Needed   
...            ...              ...                           ...   
85902      Inbound   Refund Related                Refund Enquiry   
85903      Inbound    Order Related        Seller Cancelled Order   
85904      Inbound    Order Related          Order status enquiry   
85905      Inbound         Feedback      UnProfessional Behaviour   
85906      Inbound          Returns        Reverse Pickup Enquiry   

        issue_reported_at     issue_responded survey_response_date  \
0     2023-08-01 11:13:00 2023-08-01 11:47:00           2023-08-01   
1     2023-08-01 12:52:00 2023-08-01 12:54:00           2023-08-01   
2     2023-08-01 20:16:00 2023-08-01 20:38:00           2023-08-01   
3     2023-08-01 20:56:00 2023-08-01 21:16:00           2023-08-01   
4     2023-08-01 10:30:00 2023-08-01 10:32:00           2023-08-01   
...                   ...                 ...                  ...   
85902 2023-08-30 23:20:00 2023-08-31 07:22:00           2023-08-31   
85903 2023-08-31 08:15:00 2023-08-31 08:17:00           2023-08-31   
85904 2023-08-31 18:57:00 2023-08-31 19:02:00           2023-08-31   
85905 2023-08-31 19:59:00 2023-08-31 20:00:00           2023-08-31   
85906 2023-08-31 23:36:00 2023-08-31 23:37:00           2023-08-31   

      customer_city product_category  item_price  connected_handling_time  \
0           Unknown          Unknown       979.0                    427.0   
1           Unknown          Unknown       979.0                    427.0   
2           Unknown          Unknown       979.0                    427.0   
3           Unknown          Unknown       979.0                    427.0   
4           Unknown          Unknown       979.0                    427.0   
...             ...              ...         ...                      ...   
85902       Unknown          Unknown       979.0                    427.0   
85903       Unknown          Unknown       979.0                    427.0   
85904       Unknown          Unknown       979.0                    427.0   
85905       Unknown          Unknown       979.0                    427.0   
85906       Unknown          Unknown       979.0                    427.0   

                agent_name      supervisor          manager    tenure_bucket  \
0         Richard Buchanan     Mason Gupta  Jennifer Nguyen  On Job Training   
1            Vicki Collins       Dylan Kim      Michael Lee              >90   
2             Duane Norman    Jackson Park      William Kim  On Job Training   
3           Patrick Flores     Olivia Wang       John Smith              >90   
4      Christopher Sanchez  Austin Johnson      Michael Lee             0-30   
...                    ...             ...              ...              ...   
85902         Brandon Leon       Ethan Tan      William Kim  On Job Training   
85903         Linda Foster      Noah Patel       Emily Chen              >90   
85904    Kimberly Martinez     Aiden Patel       Olivia Tan  On Job Training   
85905        Daniel Martin   Olivia Suzuki       Olivia Tan              >90   
85906     Elizabeth Guerra    Nathan Patel  Jennifer Nguyen  On Job Training   

      agent_shift  csat_score  
0         Morning           5  
1         Morning           5  
2         Evening           5  
3         Evening           5  
4         Morning           5  
...           ...         ...  
85902     Morning           4  
85903     Morning           5  
85904     Evening           5  
85905     Morning           4  
85906     Evening           5  

[85907 rows x 16 columns]

# VISUALIZATION